### PR TITLE
Remove use of global variables for configuration

### PR DESCRIPTION
**Pull request by metcalf**
In several places, Zeus propogates configuration by mutating global variables. This makes packages harder to test and interfaces less clear. This PR replaces all mutable global variables with explicit local arguments.

### CODE REVIEW

File: go/filemonitor/filemonitor.go [hunk 1]
```diff
@@ -5,7 +5,7 @@ import (
 	"time"
 )
 
-var FileChangeDelay = 300 * time.Millisecond
+const defaultFileChangeDelay = 300 * time.Millisecond
```
Comment (ptarjan) [inline go/filemonitor/filemonitor.go]: I'm not in love with this being in two places

File: go/filemonitor/filelistener_test.go [hunk 2]
```diff
@@ -25,7 +25,7 @@ func TestFileListener(t *testing.T) {
 	}
 
 	slog.SetTraceLogger(slog.NewTraceLogger(os.Stderr))
-	fl := filemonitor.NewFileListener(ln)
+	fl := filemonitor.NewFileListener(50*time.Millisecond, ln)
```
Comment (ptarjan) [inline go/filemonitor/filelistener_test.go]: a new default? Maybe extract out to a `const`.

File: go/filemonitor/filemonitor_test.go [hunk 3]
```diff
@@ -47,7 +47,7 @@ func TestFileMonitor(t *testing.T) {
 		t.Fatal(err)
 	}
 
-	fm, err := filemonitor.NewFileMonitor()
+	fm, err := filemonitor.NewFileMonitor(100 * time.Millisecond)
```
Comment (ptarjan) [inline go/filemonitor/filemonitor_test.go]: all these random numbers are making me antsy

**Comment by ptarjan** [comment]
lgtm

**Comment by metcalf** [comment]
Yeaa.... all those magic numbers were micro-optimizations that made tests run 1 second faster. Totally not worth the potential for confusion. I brought back the public `DefaultFileChangeDelay` and use that consistently.

**Pull request merged**